ddsketch/store/collapsing_lowest_dense_store.go

```diff
@@ -122,53 +145,47 @@ func (s *CollapsingLowestDenseStore) MergeWith(other Store) {
 		}
 		return
 	}
-	if s.count == 0 {
-		s.copy(o)
-		return
+	if o.minIndex < s.minIndex || o.maxIndex > s.maxIndex {
+		s.extendRange(o.minIndex, o.maxIndex)
 	}
-	s.growRight(o.maxIndex)
-	s.growLeft(o.minIndex)
-	for i := max(s.minIndex, o.minIndex); i <= min(s.maxIndex, o.maxIndex); i++ {
-		s.bins[i-s.minIndex] += o.bins[i-o.minIndex]
+	idx := o.minIndex
+	for ; idx < s.minIndex && idx <= o.maxIndex; idx++ {
+		s.bins[0] += o.bins[idx-o.offset]
 	}
-	var n float64
-	for i := o.minIndex; i <= min(s.minIndex-1, o.maxIndex); i++ {
-		n += o.bins[i-o.minIndex]
+	for ; idx < o.maxIndex; idx++ {
+		s.bins[idx-s.offset] += o.bins[idx-o.offset]
+	}
+	// This is a separate test so that the comparison in the previous loop is strict (<) and handles
+	// store.maxIndex = Integer.MAX_VALUE.
+	if idx == o.maxIndex {
+		s.bins[idx-s.offset] += o.bins[idx-o.offset]
```

Comment:
Why is this separate condition necessary?

Reply:
With `idx <= o.maxIndex` in the previous loop, if `o.maxIndex` has the largest possible value, it would overflow and continue the iteration over negative values (we've had such issues in practice in Java).